src/main/java/it/gov/pagopa/arc/exception/ArcExceptionHandler.java

```diff
@@ -0,0 +1,75 @@
+import org.springframework.core.annotation.Order;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MissingRequestHeaderException;
+import org.springframework.web.bind.MissingServletRequestParameterException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+@Slf4j
+@Order(Ordered.HIGHEST_PRECEDENCE)
+public class ArcExceptionHandler {
+    @ExceptionHandler(MissingRequestHeaderException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ArcErrorDTO handleMissingRequestHeaderExceptions(
+            MissingRequestHeaderException ex, HttpServletRequest request) {
+
+        String message = ex.getMessage();
+
+        log.info("A MissingRequestHeaderException occurred handling request {}: HttpStatus 400 - {}",
+                ArcExceptionHandler.getRequestDetails(request), message);
+        log.debug("Something went wrong handling request", ex);
+        return new ArcErrorDTO(ArcErrorDTO.ErrorEnum.INVALID_REQUEST, message);
+    }
+    @ExceptionHandler(MissingServletRequestParameterException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ArcErrorDTO handleMissingServletRequestParameterException(
+            MissingServletRequestParameterException ex, HttpServletRequest request) {
+
+        String message = ex.getMessage();
+
+        log.info("A MissingServletRequestParameterException occurred handling request {}: HttpStatus 400 - {}",
+                ArcExceptionHandler.getRequestDetails(request), message);
+        log.debug("Something went wrong handling request", ex);
+        return new ArcErrorDTO(ArcErrorDTO.ErrorEnum.INVALID_REQUEST, message);
+    }
+    @ExceptionHandler(UnauthorizedException.class)
+    public ResponseEntity<ArcErrorDTO> handleUnauthorizedException(RuntimeException ex, HttpServletRequest request){
+        return handleArcErrorException(ex, request, HttpStatus.UNAUTHORIZED, ArcErrorDTO.ErrorEnum.UNAUTHORIZED);
+    }
+    @ExceptionHandler(TooManyRequestException.class)
+    public ResponseEntity<ArcErrorDTO> handleTooManyRequestException(RuntimeException ex, HttpServletRequest request){
+        return handleArcErrorException(ex, request, HttpStatus.TOO_MANY_REQUESTS, ArcErrorDTO.ErrorEnum.TOO_MANY_REQUEST);
+    }
+    @ExceptionHandler(InternalServerException.class)
+    public ResponseEntity<ArcErrorDTO> handleInternalServerException(RuntimeException ex, HttpServletRequest request){
+        return handleArcErrorException(ex, request, HttpStatus.INTERNAL_SERVER_ERROR, ArcErrorDTO.ErrorEnum.INTERNAL_SERVER_ERROR);
+    }
+    private static ResponseEntity<ArcErrorDTO> handleArcErrorException(RuntimeException ex, HttpServletRequest request, HttpStatus httpStatus, ArcErrorDTO.ErrorEnum errorEnum) {
```

Comment:
is the reference to 'Arc' necessary?

Reply:
it's just a name, it's not necessary, I can remove the "Arc" reference from the class name